Jg/alp 130 link robot node with lidar subscribing node

juliagranato opened the pull request:
edit: bugs are gone and it works now!

juliagranato committed:
JG/ALP-130: Restructured Lidar Subscriber to incorporate with create3-fastslam.cpp

juliagranato committed:
JG/ALP-130: Restructured Lidar Subscriber to incorporate with create3-fastslam.cpp (changes actually incorporated)

juliagranato committed:
JG/ALP-130 Undefined Reference Error with laserscan_to_landmarks

juliagranato committed:
JGALP-130: Restructured CMakeLists.txt to successfully incorporate lidar_subscriber.cpp with create3-fastslam.cpp

juliagranato committed:
JGALP-130: Restructured CMakeLists.txt to successfully incorporate lidar_subscriber.cpp with create3-fastslam.cpp

WillWu88 commented on CMakeLists.txt:
```diff
@@ -1,10 +1,11 @@
 
+# Add the FastSLAM directory
 add_subdirectory(FastSLAM)
+add_library(LidarSubscriberLib src/lidar_subscriber.cpp)
```
I don't know if it's best practice to add a single source file that's meant to be run to a separate library. For now we can keep it, but this should be refactored in the future.

juliagranato replied:
I totally agree. I was really struggling to properly link the header and source files and tried a bunch of things. For some reason that was the only thing that worked so I left it

WillWu88 commented on include/lidar_subscriber.h:
Why are there two files that does the same thing with slightly different names?

juliagranato replied:
Not sure... there is only one now

WillWu88 commented on src/build/.built_by:
Please delete this and the "install" and "build" folder. Looks like you invoked colcon in the wrong directory.

WillWu88 commented on src/lidar-subscriber.h:
Why is there another header file here that basically does the same thing as the last two?

WillWu88 commented on src/lidar_subscriber.cpp:
```diff
@@ -96,46 +109,11 @@ std::queue<Observation2D> LidarSubscriber::laserscan_to_landmarks(const sensor_m
-  RCLCPP_DEBUG(this->get_logger(), "%u %f %f\n", landmarks.size(), landmarks.front().range_m, landmarks.front().bearing_rad);
-  #ifdef VISUALIZE_LANDMARKS
+
+  std::cout<<landmarks.size()<<"\n";
```
Please replace this with the ROS logging framework

juliagranato replied:
Done!

WillWu88 commented on src/l:
Is this made and included by mistake?

juliagranato replied:
I believe so. It has been deleted

WillWu88 reviewed (changes requested):
Please clean the source directory and your changes... way too much junk artifacts and files included here. Thanks

juliagranato committed:
JG/ALP-130 Clean src and include directories

Removed unnecessary files from src and include directories and created a launch file so that the rplidar, laser filter, and create3 node can all be launch from one command

juliagranato committed:
JG/ALP-130: Removed unnecessary Files

I learned you have to git add -A for git to learn that files have been deleted

juliagranato committed:
Revert "JG/ALP-130: Removed unnecessary Files"

This reverts commit 65b9326e840f788d46de7ff993eefb92cc813ac2.

juliagranato committed:
JG/ALP-130: Properly deleted extra garbage without changing FastSLAM
